hybrid/compartment_examples/call_sentry/main.c

```diff
@@ -0,0 +1,88 @@
+/***
+ * This example showcases two particular actions to help compartmentalization:
+ *      - reducing the bounds of the PCC (shared.s:82)
+ *      - calling a sentry from within a bounded compartment (shared.s:114)
+ *
+ * One difficulty, and capability leak, is how to return from the sentry
+ * function back to the caller with correct capabilities (i.e., PCC). In this
+ * example, we use `clr`, and manually return to it (main.c:49). However,
+ * having the `clr` available to the sentry function means it could access
+ * capabilities it should not be privy to (e.g., if the bounds of the sentry
+ * are set to only cover the specific function. In this example, this is
+ * further aggravated by the jump between assembly and `C`: the `C` function is
```

Comment:
I'm not sure how the jump between assembly and C aggravates things? Making use of `lr` isn't a (capability) problem as such. I agree that it's slightly annoying/awkward to have to transition from a capability-aware thing (in assembly) and a non-capability-aware thing (in C), but I'm not sure I can see this adding security issues?

Reply:
I didn't mean "further aggravat[ing]" the security stuff, more that horrendous `add sp` instruction. I can reword it if it sounds out of place.